### PR TITLE
DICOM Store support

### PR DESCRIPTION
**Pull request by loafoe**

### CODE REVIEW

File: dicom/config_service.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package dicom
```
Comment (gjkamstra) [inline dicom/config_service.go:1]: This file seems like a "base class", it contains the struct definitions for other (but not all) api's, but also contains the SetCrdServiceAccount. I would expect a config_service_cdr_account file, only containing these methods. I would also just move all struct definitions to their usage file.

File: dicom/config_service_import_service.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package dicom
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type ImportService struct {
+	ID      string `json:"id,omitempty"`
```
Comment (gjkamstra) [inline dicom/config_service_import_service.go:9]: There is also a description field

File: dicom/config_service_object_stores.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+// CredsServiceAccess
+type CredsServiceAccess struct {
+	ProductKey     string `json:"productKey"`
```
Comment (gjkamstra) [inline dicom/config_service_object_stores.go:12]: I am missing the endPoint param

**Review by gjkamstra — COMMENTED** [review]
Great work, will really help in combiantion with Terraform provider and module.

File: dicom/client.go [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+		return nil, err
+	}
+	if config.DebugLog != "" {
+		var err error
+		c.debugFile, err = os.OpenFile(config.DebugLog, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+		if err != nil {
+			c.debugFile = nil
+		}
+	}
+	ma, err := jsonformat.NewMarshaller(false, "", "", jsonformat.STU3)
```
[the remark this answers is not in the document]
Comment (loafoe) [reply dicom/client.go:79]: This was in anticipation of CDR related calls, which have not materialized. So I'll remove these in an upcoming release.

File: dicom/client.go [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+func (o *Client) GetWADOURL() string {
+	return o.replaceConfigWith("wado")
+}
+
+func (o *Client) GetQIDOURL() string {
+	return o.replaceConfigWith("qido")
+}
+
+func (o *Client) replaceConfigWith(svc string) string {
+	return strings.Replace(o.config.DICOMConfigURL, "config", svc, 1)
```
[the remark this answers is not in the document]
Comment (loafoe) [reply dicom/client.go:114]: This was added after discussion with the DICOM team. I actually requested to have an API endpoint which returns these values from the config API itself. Until/If that is in place the above mechanism is guaranteed to work, at least that is what was promised, haha

One way to make it bit more robust is to parse the URL and only apply the replace operation  only to the hostname.

File: dicom/client_test.go [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+		if r.Method != "POST" {
+			t.Errorf("Expected ‘POST’ request")
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		_, _ = io.WriteString(w, `{
+    "scope": "mail tdr.contract tdr.dataitem",
```
[the remark this answers is not in the document]
Comment (loafoe) [reply dicom/client_test.go:71]: copy / paste from another test :)

File: dicom/config_service_import_service.go (hunk 2 above)
Comment (loafoe) [reply dicom/config_service_import_service.go:9]: Ah, this one has bitten me before. the HSDP documentation sometimes add a `description:` text that looks very much like an actual Model field. This is also the case for the `configData` description. https://www.hsdp.io/documentation/dicom-store/api-documents/configuration-api#/ImportService/CreateOrUpdateImportService

File: dicom/config_service_object_stores.go (hunk 3 above)
Comment (loafoe) [reply dicom/config_service_object_stores.go:12]: Good catch, added in eda33dd together with `ServiceAccount.Name` which was also missing

File: dicom/config_service.go (hunk 1 above)
Comment (loafoe) [reply dicom/config_service.go:1]: The `dicom` package is the first one where I pervasively grouped structs and member functions in separate files. I was a bit sloppy here and that should be fixed in 4907c0e . By splitting it up it also uncovered missing test code for the `SetCDRServiceAccount` code path, thanks! Appreciate the very thorough code review 👍🏻

File: dicom/config_service_import_service.go (hunk 2 above)
Comment (gjkamstra) [reply dicom/config_service_import_service.go:9]: I know

**Pull request merged**